staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -132,7 +127,7 @@ func (c *tlsTransportCache) get(config *Config) (http.RoundTripper, error) {
 
 	if canCache {
 		// Cache a single transport for these options
-		c.transports[key] = transport
+		c.transports.Add(key, transport, 19*time.Hour)
```

Comment:
do we know the case of one transport used by one client that will expire?
maybe we can just use an lru cache 	"github.com/golang/groupcache/lru" without expiration